src/axom/quest/MarchingCubes.hpp

```diff
@@ -137,227 +156,213 @@ class MarchingCubes
-   *
-   * Some data from \a dom may be cached by the constructor.
-   * Any change to it after the constructor leads to undefined behavior.
-   *
-   * The mesh coordinates should be stored contiguously.  See
-   * conduit::blueprint::is_contiguous().  In the future, this
-   * requirement may be relaxed, possibly at the cost of a
-   * transformation and storage of the temporary contiguous layout.
-   */
-  MarchingCubesSingleDomain(RuntimePolicy runtimePolicy,
-                            const conduit::Node &dom,
-                            const std::string &topologyName,
-                            const std::string &maskfield);
-
-  void setDataParallelism(MarchingCubesDataParallelism &dataPar)
+    Memory space of data corresponds to allocator set in the constructor.
+  */
+  axom::ArrayView<const double, 2> getContourNodeCoords() const
   {
-    m_dataParallelism = dataPar;
+    return m_facetNodeCoords.view();
   }
 
   /*!
-    @brief Specify the field containing the nodal scalar function
-    in the input mesh.
-    \param [in] fcnField Name of node-based scalar function values.
-  */
-  void setFunctionField(const std::string &fcnField);
+    @brief Return view of parent cell indices Array.
 
-  /*!
-    @brief Get the Blueprint domain id specified in \a state/domain_id
-    if it is provided, or use the given default if not provided.
+    The buffer size is getContourCellCount().  The parent ID is the
+    column-major ordered flat index of the cell in the parent domain
+    (see ArrayIndexer), not counting ghost cells.
+
+    Memory space of data corresponds to allocator set in the constructor.
   */
-  int getDomainId(int defaultId) const;
+  axom::ArrayView<const axom::IndexType> getContourFacetParents() const
+  {
+    return m_facetParentIds.view();
+  }
 
   /*!
-   * \brief Compute the isocontour.
-   *
-   * \param [in] contourVal isocontour value
-   *
-   * Compute isocontour using the marching cubes algorithm.
-   * To get the isocontour mesh, use populateContourMesh.
-   */
-  void computeIsocontour(double contourVal = 0.0);
+    @brief Return view of parent domain indices Array.
+    @param allocatorID Allocator id for the output data.  If omitted,
+           use the id set in the constructor.
 
-  //!@brief Get number of cells in the generated contour mesh.
-  axom::IndexType getContourCellCount() const
+    The buffer size is getContourCellCount().
+  */
+  axom::ArrayView<const axom::IndexType> getContourFacetDomainIds() const
   {
-    SLIC_ASSERT_MSG(
-      m_impl,
-      "There is no contour mesh until you call computeIsocontour()");
-    axom::IndexType cellCount = m_impl->getContourCellCount();
-    return cellCount;
+    return m_facetDomainIds.view();
   }
 
-  //!@brief Get number of nodes in the generated contour mesh.
-  axom::IndexType getContourNodeCount() const
+  /*!
+    @brief Give caller posession of the contour data.
+
+    This efficiently turns the generated contour data to the caller,
```

Comment:
Typo: you probably meant "returns the generated contour data"

Reply:
Actually, it's not a typo, but it's probably confusing.  Will rephrase to "This efficiently gives the generated contour data to the caller..."